src/binding-hash.h

```diff
@@ -18,34 +18,47 @@
 #define WABT_BINDING_HASH_H_
 
 #include "common.h"
-#include "vector.h"
+
+#include <string>
+#include <vector>
+#include <unordered_map>
 
 namespace wabt {
 
 struct Binding {
+  explicit Binding(int index) : index(index) {
+    WABT_ZERO_MEMORY(loc);
+  }
+  Binding(const Location& loc, int index) : loc(loc), index(index) {}
+
   Location loc;
-  StringSlice name;
   int index;
 };
 
-struct BindingHashEntry {
-  Binding binding;
-  struct BindingHashEntry* next;
-  struct BindingHashEntry* prev; /* only valid when this entry is unused */
-};
-WABT_DEFINE_VECTOR(binding_hash_entry, BindingHashEntry);
+class BindingHash : public std::unordered_multimap<std::string, Binding> {
```

Comment:
I considered it, and it shouldn't be too bad because I think the only function I currently use from `unordered_multimap` is `emplace`. Should be a relatively simple refactor if we want to do that later.